src/awakener.rs

```diff
@@ -0,0 +1,95 @@
+use std::io;
+
+use crate::{poll, sys, Registry, Token};
+
+/// Awakener allows cross-thread waking of [`Poll`].
+///
+/// When created it will cause events with [`Ready::readable()`] and the provided
+/// `token` if [`wake`] is called, possibly from another thread.
+///
+/// # Notes
+///
+/// The `Awakener` needs to be kept alive as long as wake up notifications are
+/// required. This is due to an implementation detail where if all copies of the
+/// `Awakener` are dropped it will also drop all wake up notifications from the
+/// system queue, including wake up notifications that have been added before
+/// the `Awakener` that was dropped, resulting the [`Poll`] not being woken up.
+///
+/// Only a single `Awakener` should active per [`Poll`], if multiple threads
+/// need access to the `Awakener` it can be shared via for example an `Arc`.
+/// What happens if multiple `Awakener`s are registered with the same `Poll` is
```

Comment:
Why is it undefined? If they are registered with different tokens, it should be OK?

Reply:
I put that in to allow some room for future implementations that weren't ok with multiple `Awakener`s, but currently it's fine on all platforms I think

Reply:
I can remove it if you want.

Reply:
I'm just wondering about it. I was thinking, since tokens are exposed, you could have multiple wakers that have different significance.

Reply:
Want me to remove it?

Reply:
I've tried adding a test for this and it doesn't work on kqueue platforms currently.

Reply:
I could try and fix that.